nucypher/utilities/networking.py

```diff
@@ -200,4 +204,53 @@ def determine_external_ip_address(
     # complete failure!
     if not rest_host:
         raise UnknownIPAddress('External IP address detection failed')
+
     return rest_host
+
+
+def _resolve_ipv4(ip: str) -> Optional[IPv4Address]:
+    """Resolve an IP address to IPv4 if required and possible."""
+    try:
+        ip = ip_address(ip.strip())
+    except AddressValueError:
+        return None
+    if isinstance(ip, IPv6Address) and ip.ipv4_mapped:
+        return ip.ipv4_mapped
+    elif isinstance(ip, IPv4Address):
+        return ip
```

Comment:
Looks like `ipv4_mapped` can return `None` so should we just do this (I should have noticed this earlier), then all our cases are handled.

```suggestion
    if isinstance(ip, IPv6Address):
        return ip.ipv4_mapped
    # if not ipv6 then it is ipv4 already
    return ip
```

Reply:
✔️